### ISSUE TITLE
Add support for table option in ALTER TABLE statements for MySQL/MariaDB

### ISSUE THREAD

**Issue by pprkut** [sqlfluff/sqlfluff#6983]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following query yields parsing errors:

```sql
ALTER TABLE `foo`
ENGINE = 'InnoDB'
AUTO_INCREMENT = 1
AVG_ROW_LENGTH = 10
DEFAULT CHARACTER SET = 'utf8_unicode_ci'
CHECKSUM = 1
DEFAULT COLLATE = 'utf8mb4_unicode_ci'
COMMENT = 'comment'
CONNECTION = 'connection_string'
DELAY_KEY_WRITE = 0
ENCRYPTED = NO
ENCRYPTION_KEY_ID = 1234
IETF_QUOTES = YES
INDEX DIRECTORY = 'path/to/dir'
INSERT_METHOD = LAST
KEY_BLOCK_SIZE = 1024
MAX_ROWS = 100000
MIN_ROWS = 1
PACK_KEYS = 1
PAGE_CHECKSUM = 1
PAGE_COMPRESSED = 0
PAGE_COMPRESSION_LEVEL = 9
PASSWORD = 'password'
ROW_FORMAT = DYNAMIC
SEQUENCE = 1
STATS_AUTO_RECALC = 1
STATS_PERSISTENT = 1
STATS_SAMPLE_PAGES = 4
TABLESPACE tablespace_name
TRANSACTIONAL = 1
UNION = (`tbl1`,`tbl2`)
WITH SYSTEM VERSIONING;
```

### Expected Behaviour

No parsing errors

### Observed Behaviour

```
== [test.sql] FAIL
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section:
                       | 'AUTO_INCREMENT = 1\nAVG_ROW_LENGTH = 10\nD...'
WARNING: Parsing errors found and dialect is set to 'mariadb'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

`sqlfluff lint --dialect=mariadb test.sql`

### Dialect

Happens with both `mysql` and `mariadb`

### Version

`sqlfluff, version 3.4.0`
`Python 3.12.11``

### Configuration

No special config

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
